app/docker/views/containers/edit/containerController.js

```diff
@@ -192,40 +192,116 @@ function ($q, $scope, $state, $transition$, $filter, Commit, ContainerHelper, Co
     var container = $scope.container;
     var config = ContainerHelper.configFromContainer(container.Model);
     $scope.state.recreateContainerInProgress = true;
-    ContainerService.remove(container, true)
-    .then(function success() {
+    var isRunning = container.State.Running;
+
+    return stopContainerIfNeeded()
+      .then(renameContainer)
+      .then(pullImageIfNeeded)
+      .then(createContainer)
+      .then(connectContainerToOtherNetworks)
+      .then(startContainerIfNeeded)
+      .then(createResourceControlIFNeeded)
+      .then(deleteOldContainer)
+      .then(notifyAndChangeView)
+      .catch(onError);
+
+    function stopContainerIfNeeded() {
+      if (!isRunning) {
+        return $q.when();
+      }
+      return ContainerService.stopContainer(container.Id);
+    }
+
+    function renameContainer() {
+      return ContainerService.renameContainer(container.Id, container.Name + '-old');
+    }
+
+    function pullImageIfNeeded() {
+      if (!pullImage) {
+        return $q.when();
+      }
+      return getRegistery().then(function pullImage(containerRegistery) {
+        return ImageService.pullImage(container.Config.Image, containerRegistery, true);
+      });
+    }
+
+    function getRegistery() {
+      // get container registery
       return RegistryService.retrieveRegistryFromRepository(container.Config.Image);
-    })
-    .then(function success(data) {
-      return $q.when(!pullImage || ImageService.pullImage(container.Config.Image, data, true));
-    })
-    .then(function success() {
-      return ContainerService.createAndStartContainer(config);
-    })
-    .then(function success(data) {
-      if (!container.ResourceControl) {
-        return true;
-      } else {
-        var containerIdentifier = data.Id;
-        var resourceControl = container.ResourceControl;
-        var users = resourceControl.UserAccesses.map(function(u) {
-          return u.UserId;
-        });
-        var teams = resourceControl.TeamAccesses.map(function(t) {
-          return t.TeamId;
+    }
+
+    function createContainer() {
+      // leave only one network on the container (save the others for later)
+      var networks = config.NetworkingConfig.EndpointsConfig;
+      var networksNames = Object.keys(networks);
+      if (networksNames.length > 1) {
+        config.NetworkingConfig.EndpointsConfig = {};
+        config.NetworkingConfig.EndpointsConfig[networksNames[0]] = networks[0];
+      }
+      // create container
+      return $q.all([ContainerService.createContainer(config), networks]);
+      // return ContainerService.createContainer(config);
+    }
+
+    function connectContainerToOtherNetworks(data) {
+      var newContainer = data[0];
+      var networks = data[1];
+      var networksNames = Object.keys(networks);
+      return $q
+        .all(networksNames.map(function connectToNetwork(name) {
```

Comment:
I think that the `NetworkService.connectContainer` array should be created before calling `$q.all`, it will be easier to read.